go/integration_test/interpreter/stress_test.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package interpreter_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/Fantom-foundation/Tosca/go/interpreter/lfvm"
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+	"github.com/Fantom-foundation/Tosca/go/tosca/vm"
+	"golang.org/x/sync/errgroup"
+)
+
+func BenchmarkParallelStressTests(b *testing.B) {
+	codes := map[string][]byte{
+		"smallTransaction": getStressBenchmark_SmallTransaction(),
+		"largeCodes":       getStressBenchmark_LargeCodes(),
+		"stressHashCache":  getStressBenchmark_StressHashCache(),
+		"noCodeReUsage":    getStressBenchmark_NoCodeReUsage(),
+	}
+
+	for _, interpreterName := range getAllInterpreterVariantsForTests() {
+		interpreter, err := tosca.NewInterpreter(interpreterName)
+		if err != nil {
+			b.Fatalf("failed to load %s with error: %v", interpreterName, err)
+		}
+		for _, numCalls := range []int{1, 10, 100, 1000, 10000} {
+			for codeName, code := range codes {
+				b.Run(fmt.Sprintf("%s-%s-numCalls=%d", interpreterName, codeName, numCalls), func(b *testing.B) {
+					hash := lfvm.Keccak256(code)
+					errs, _ := errgroup.WithContext(context.Background())
+					errs.SetLimit(-1) // set limit of max goroutines to unlimited
+
+					start := time.Now()
+					for i := 0; i < numCalls; i++ {
+						errs.Go(func() error {
+
+							if codeName == "noCodeReUsage" {
+								// code has to be different for each call
+								code = append(code, byte(vm.PUSH4))
+								code = append(code, uint32ToBytes(uint32(i))...)
+								code = append(code, byte(vm.STOP)) // stop
+								hash = lfvm.Keccak256(code)
```

Comment:
There should be a race condition on this benchmark ... all parallel go-routines are using the same code and hash instance.

Also, is the intention that the code is gradually extended or should just the value to be pushed be replaced?

Reply:
Yes, I didn't notice the race condition when I refactored the different tests into a single one, fixed now.
I planed to simply extend it, as it should not matter what the code does it just needs to be different.